src/main/groovy/com/github/jrubygradle/internal/GemVerion.groovy

```diff
@@ -0,0 +1,122 @@
+package com.github.jrubygradle.internal
+
+class GemVersion {
+
+    private String low
+    private String high
+    private prefix = '['
+    private postfix = ']'
+  
+    private GemVersion(String pre, String l, String h, String post) {
+        prefix = pre
+        low = l
+        high = h
+        postfix = post
+    }
+
+    GemVersion(String version) {
+        if (version.contains( '+' ) ) {
+            low = version.replaceFirst(/\.\+/, '.0').replaceFirst(/\+/, '')
+            high = version.replaceFirst(/\.\+/, '.99999').replaceFirst(/[0-9]+\+/, '99999')
+        }
+        else if (version.contains( '[' ) || version.contains( '(' ) ||
```

Comment:
string literals will be interned - see http://stackoverflow.com/questions/3297867/difference-between-string-object-and-string-literal

assuming groovy does the same interning of string literals. not sure what groovy does with those regex expressions. will use java Pattern.compile instead